docs/github-actions.md

```diff
@@ -0,0 +1,174 @@
+1. Analytics engineers will issue a pull request on GitHub
+2. The GitHub Action will use Databricks CLI to [create and trigger a one-time run](https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsRunsSubmit).
+3. The Job will run `dbt test`
+4. SQL generated by dbt will run on a Databricks Cluster. For ease of development, we will use an All Purpose cluster which is already running. You can run dbt on an Automated Cluster instead.
+
+## Prerequisites
+- Access to a Databricks workspace
+- Ability to create a Personal Access Token (PAT)
+- A fork of the [jaffle_shop](https://github.com/dbt-labs/jaffle_shop) demo project. You can, alternatively, follow along with your own dbt project.
+
+## Create GitHub Action
+1. Create a `.github/workflows` directory
+
+```
+$ mkdir -p .github/workflows
+$ cd .github/workflows
+```
+
+2. Create a `requirements.txt` file and paste the following, which specifies the version of the Databricks CLI to install
+
+```nofmt
+databricks-cli
+```
+
+3. Create a file named `job.json` and paste the following content:
+
+```json
+{
+    "run_name": "jaffle_shop_ci",
+    "tasks": [
+        {
+            "task_key": "jaffle_shop_tests",
+            "dbt_task": {
+                "commands": [
+                    "dbt debug",
+                    "dbt test"
+                ]
+            },
+            "existing_cluster_id": "_EXISTING_CLUSTER_ID_",
+            "libraries": [
+                {
+                    "pypi": {
+                        "package": "dbt-core>=1.0.0,<2.0.0"
+                    }
+                },
+                {
+                    "pypi": {
+                        "package": "dbt-databricks>=1.0.0,<2.0.0"
+                    }
+                }
+            ]
+        }
+    ],
+    "git_source": {
+        "git_url": "https://github.com/dbt/jaffle_shop",
+        "git_provider": "gitHub",
+        "git_branch": "_GITHUB_BRANCH_"
+    }
+}
+```
+
+Note that we have two template placeholders in the JSON document which will get replaced by the GitHub Action:
+- `_EXISTING_CLUSTER_ID_` is the Cluster ID of an existing All Purpose Cluster
+- `_GITHUB_BRANCH_` is the Git branch of our project we want to deploy
+
+5. Create a file called `main.yml` and paste the GitHub Action definition in it. This GitHub Action will check out your dbt repo, install the Databricks CLI, submit the job spec you defined above using the Databricks CLI, and check until the job completes or fails.
+
+
+```yml
+name: Databricks Job
+
+on:
+  workflow_dispatch:
+    branches: [main]
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  DATABRICKS_TOKEN: ${{ secrets.DATABRICKS_TOKEN }}
+  DATABRICKS_HOST: ${{ secrets.DATABRICKS_HOST }}
+  EXISTING_CLUSTER_ID: ${{ secrets.DATABRICKS_CLUSTER_ID }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: your-workspace
+    strategy:
+      matrix:
+        python-version: ["pypy-3.7"]
```

Comment:
3.7 should work fine but I think we mostly test dbt with 3.8 so I'd recommend that version.